presto-main/src/main/java/com/facebook/presto/sql/analyzer/MaterializedViewQueryOptimizer.java

```diff
@@ -94,55 +117,268 @@
-    private final Table materializedView;
-    private final Query materializedViewQuery;
     private final LogicalRowExpressions logicalRowExpressions;
-
-    private MaterializedViewInfo materializedViewInfo;
-    private Optional<Identifier> removablePrefix = Optional.empty();
-    private Optional<Set<Expression>> expressionsInGroupBy = Optional.empty();
+    private final Map<QualifiedObjectName, List<ConnectorMaterializedViewDefinition>> baseTableToMaterializedViews;
 
     public MaterializedViewQueryOptimizer(
             Metadata metadata,
             Session session,
             SqlParser sqlParser,
             AccessControl accessControl,
             RowExpressionDomainTranslator domainTranslator,
-            Table materializedView,
-            Query materializedViewQuery)
+            Map<QualifiedObjectName, List<ConnectorMaterializedViewDefinition>> baseTableToMaterializedViews)
     {
         this.metadata = requireNonNull(metadata, "metadata is null");
         this.session = requireNonNull(session, "session is null");
         this.sqlParser = requireNonNull(sqlParser, "sql parser is null");
         this.accessControl = requireNonNull(accessControl, "access control is null");
         this.domainTranslator = requireNonNull(domainTranslator, "row expression domain translator is null");
-        this.materializedView = requireNonNull(materializedView, "materialized view is null");
-        this.materializedViewQuery = requireNonNull(materializedViewQuery, "materialized view query is null");
+        this.baseTableToMaterializedViews = requireNonNull(baseTableToMaterializedViews, "base table to materialized views is null");
         FunctionAndTypeManager functionAndTypeManager = metadata.getFunctionAndTypeManager();
         logicalRowExpressions = new LogicalRowExpressions(
                 new RowExpressionDeterminismEvaluator(functionAndTypeManager),
                 new FunctionResolution(functionAndTypeManager),
                 functionAndTypeManager);
     }
 
-    public Node rewrite(Node node)
+    @Override
+    protected Node visitQuerySpecification(QuerySpecification node, Void context)
     {
-        // TODO: Implement ways to handle non-optimizable query without throw/catch. https://github.com/prestodb/presto/issues/16541
-        try {
-            MaterializedViewInformationExtractor materializedViewInformationExtractor = new MaterializedViewInformationExtractor();
-            materializedViewInformationExtractor.process(materializedViewQuery);
-            materializedViewInfo = materializedViewInformationExtractor.getMaterializedViewInfo();
-            return new QuerySpecificationRewriter().process(node);
-        }
-        catch (Exception ex) {
-            logger.warn("Failed to rewrite query with materialized view with following exception: %s", ex.getMessage());
+        if (!node.getFrom().isPresent()) {
+            log.warn("Query with no from clause are not optimized by materialized view");
```

Comment:
Lets remove all log.warns, we want to enable this for all queries eventually, and will just spam